public/js/components/Frames.js

```diff
@@ -8,22 +9,24 @@ const actions = require("../actions");
 const { endTruncateStr } = require("../utils/utils");
 const { getFilename } = require("../utils/source");
 const { getFrames, getSelectedFrame, getSource } = require("../selectors");
+const { List } = require("immutable");
+const classNames = require("classnames");
 
 if (typeof window == "object") {
   require("./Frames.css");
 }
 
 const NUM_FRAMES_SHOWN = 7;
 
-function renderFrameTitle(frame) {
-  return div({ className: "title" }, endTruncateStr(frame.displayName, 40));
+function renderFrameTitle({ displayName }: Object) {
```

Comment:
Got that to work when `Frame` is a type, but now the type declaration is really long:
```javascript
function renderFrameLocation({ source, location }: {source: Source, location: Location}) {
```
Ultimately I'm not sure it's worth the additional annotations. But at least `{ source, location}: Frame` is more descriptive.